libfwupdplugin/fu-udev-device.c

```diff
@@ -605,6 +605,15 @@ fu_udev_device_get_open_flags(FuUdevDevice *self)
 	return priv->open_flags;
 }
 
+static void
+fu_udev_device_invalidate(FuDevice *device)
+{
+	FuUdevDevice *self = FU_UDEV_DEVICE(device);
+	FuUdevDevicePrivate *priv = GET_PRIVATE(self);
+	priv->properties_valid = FALSE;
+	g_hash_table_remove_all(priv->properties);
```

Comment:
need to clear any emulation at this time too?

Reply:
That's a great question. I think it *cant* as we need to capture the bit before the call to fu_device_probe_invalidate() so we can get that far. e.g. the emulation is really "_what's provided from the hardware_" (including errors, etc) rather than "_what's desired_".